backend/application/syson-application-configuration/src/main/java/org/eclipse/syson/application/publication/SySONLibraryPublicationHandler.java

```diff
@@ -0,0 +1,258 @@
+import org.eclipse.sirius.web.domain.boundedcontexts.semanticdata.SemanticData;
+import org.eclipse.sirius.web.domain.boundedcontexts.semanticdata.services.api.ISemanticDataCreationService;
+import org.eclipse.sirius.web.domain.services.IResult;
+import org.eclipse.sirius.web.domain.services.Success;
+import org.eclipse.syson.sysml.SysmlPackage;
+import org.eclipse.syson.sysml.util.ElementUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.data.jdbc.core.mapping.AggregateReference;
+import org.springframework.stereotype.Service;
+
+/**
+ * {@link ILibraryPublicationHandler} for publishing libraries in SysON.
+ *
+ * @see SysONLibraryPublicationListener
+ * @author flatombe
+ */
+@Service
+public class SySONLibraryPublicationHandler implements ILibraryPublicationHandler {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SySONLibraryPublicationHandler.class);
+
+    private final IEditingContextSearchService editingContextSearchService;
+
+    private final ISemanticDataCreationService semanticDataCreationService;
+
+    private final IResourceToDocumentService resourceToDocumentService;
+
+    private final IProjectSemanticDataSearchService projectSemanticDataSearchService;
+
+    private final IProjectSearchService projectSearchService;
+
+    private final ILibrarySearchService librarySearchService;
+
+    public SySONLibraryPublicationHandler(final IEditingContextSearchService editingContextSearchService,
+            final ISemanticDataCreationService semanticDataCreationService,
+            final IResourceToDocumentService resourceToDocumentService,
+            final IProjectSemanticDataSearchService projectSemanticDataSearchService, final IProjectSearchService projectSearchService,
+            final ILibrarySearchService librarySearchService) {
+        this.projectSearchService = Objects.requireNonNull(projectSearchService);
+        this.editingContextSearchService = Objects.requireNonNull(editingContextSearchService);
+        this.semanticDataCreationService = Objects.requireNonNull(semanticDataCreationService);
+        this.resourceToDocumentService = Objects.requireNonNull(resourceToDocumentService);
+        this.projectSemanticDataSearchService = Objects.requireNonNull(projectSemanticDataSearchService);
+        this.librarySearchService = Objects.requireNonNull(librarySearchService);
+    }
+
+    @Override
+    public boolean canHandle(final PublishLibrariesInput input) {
+        return Objects.equals(input.publicationKind(), "Project_SysML_AllProperContents");
+    }
+
+    @Override
+    public IPayload handle(final PublishLibrariesInput input) {
+        return this.projectSearchService.findById(input.projectId()).map(project -> {
+            final Optional<IEMFEditingContext> maybeEditingContext = this.projectSemanticDataSearchService.findByProjectId(AggregateReference.to(input.projectId()))
```

Comment:
`maybeEditingContext` seems useless here you could just do it in only one fluent request